### PR TITLE
Added total_tracks to SoundCloudAlbum

### PR DESCRIPTION
**Pull request by FranklinBinns**
The property will get the total tracks of the sound cloud album via the album_tracks method. If the sound cloud album is a single it will return 1.

### CODE REVIEW

**Comment by jmbannon** [comment]
Thanks for the PR! We need to add this property to the following mixin as well:
https://github.com/jmbannon/ytdl-sub/blob/e8e1daaec29f39f5113f772004595a159fe4ce55/src/ytdl_sub/entries/variables/soundcloud_variables.py

This is the mixin used by SoundcloudAlbum to get properties and is also what is used to generate the docs found here:
https://ytdl-sub.readthedocs.io/en/latest/config/format_variables/soundcloud.html

(Sorry, I haven't documented this very well yet)

Please use Numpy docstring format. It's used in the mixin.

Lastly, we need to add this property into our unit tests. See https://github.com/jmbannon/ytdl-sub/blob/e8e1daaec29f39f5113f772004595a159fe4ce55/tests/unit/entries/test_soundcloud_entries.py

You can run the tests with `./tools/test`, ensure it shows up in the docs by running `./tools/docs`, and also run your added code through the linter with `./tools/lint`


That was a lot! I plan to make this more clear somewhere for contributors to see. If you need any help I'd be happy to hop on discord or something sometime

**Comment by FranklinBinns** [comment]
Still, working on it hope you don't mind just taking the time to understand the architecture.

File: src/ytdl_sub/entries/variables/soundcloud_variables.py [hunk 1]
```diff
@@ -25,6 +25,17 @@ def track_number_padded(self) -> str:
...
         return f"{self.track_number:02d}"
 
+    @property
+    def total_tracks(self) -> int:
+        """
+        Returns
+        -------
+        The total tracks in album. For singles, it will always be 1.
+        """
+        if self.kwargs_contains("total_tracks"):
```
Comment (jmbannon) [inline src/ytdl_sub/entries/variables/soundcloud_variables.py:35]: the kwargs here is referring to the YTDL dict returned for the soundcloud entry. I don't think total_tracks is there. We can always return 1 here just like we do with `track_number`

File: src/ytdl_sub/entries/soundcloud.py [hunk 2]
```diff
@@ -105,6 +112,7 @@ def album_tracks(self, skip_premiere_tracks: bool = True) -> List[SoundcloudAlbu
                 album=self.title,
                 album_year=self.album_year,
                 soundcloud_track=track,
+                total_tracks=self.track_count,
```
Comment (jmbannon) [inline src/ytdl_sub/entries/soundcloud.py:115]: maybe we should incorporate this into the PlaylistMetadata as `playlist_count`. Then in `total_tracks`, return `self._playlist_metadata.playlist_count`

I think the ytdl field is 'playlist_count',  i recommend using a debugger and seeing what ytdl returns in the entry dict

Comment (jmbannon) [reply src/ytdl_sub/entries/soundcloud.py:115]: That would help solve this issue: https://github.com/jmbannon/ytdl-sub/issues/9

Comment (FranklinBinns) [reply src/ytdl_sub/entries/soundcloud.py:113]: Yea that makes sense, I've incorporated it.

File: src/ytdl_sub/entries/variables/soundcloud_variables.py (hunk 1 above)
Comment (FranklinBinns) [reply src/ytdl_sub/entries/variables/soundcloud_variables.py]: Yes, didn't release what kwargs was actually doing, changed it, to that implementation.

File: src/ytdl_sub/entries/soundcloud.py (hunk 2 above)
Comment (FranklinBinns) [reply src/ytdl_sub/entries/soundcloud.py:113]: I've also launched it in a debugger and checked the entry dict, and the field is playlist_count, this is also used in the track_count property in  SoundcloudAlbum.

Comment (jmbannon) [reply src/ytdl_sub/entries/soundcloud.py:113]: Sweet looks great! Looks like the linter errored: https://github.com/jmbannon/ytdl-sub/runs/6236060213?check_suite_focus=true

One last nit: lets change the soundcloud variable name from `total_tracks` to `track_count`, that way it's near the other track variables via sorting in the documentation

Comment (FranklinBinns) [reply src/ytdl_sub/entries/soundcloud.py]: I believe I've fixed that linter error, and I've changed the variable name.

File: src/ytdl_sub/entries/soundcloud.py [hunk 3]
```diff
@@ -51,6 +51,11 @@ def album(self) -> str:
         """Returns the entry's album name, fetched from its internal album"""
         return self._album
 
+    @property
+    def total_tracks(self) -> int:
```
Comment (jmbannon) [inline src/ytdl_sub/entries/soundcloud.py:55]: need to set this to track_count

File: tests/unit/entries/test_soundcloud_entries.py [hunk 4]
```diff
@@ -23,6 +28,11 @@ def is_premiere():
     return False
 
 
+@pytest.fixture()
+def track_count():
+    return 1
```
Comment (jmbannon) [inline tests/unit/entries/test_soundcloud_entries.py:33]: dupe fixtures

Comment (jmbannon) [reply tests/unit/entries/test_soundcloud_entries.py:33]: after these two small fixes it should be good to merge

Comment (FranklinBinns) [reply tests/unit/entries/test_soundcloud_entries.py]: Fixed those two problems, thanks for being patient this has been quite the pull request at least it has been enjoyable.

Comment (jmbannon) [reply tests/unit/entries/test_soundcloud_entries.py]: No problem, thank _you_ for contributing! I hope my comments have been helpful and that you find this app useful

**Review by jmbannon — APPROVED** [review]

**Pull request merged**